### DISCUSSION TITLE
Multiple projects 1 database

### DISCUSSION THREAD

**Discussion by Jarco-dev** [prisma/prisma#12350] [Q&A] [answered]
I am kinda running into an issue, I am working on a discord bot with will have a website, the bot, and an API all running separately so 3 different projects. but how do I connect Prisma with the database in each one, do I just copy the migrations/schema over to all 3 projects, or is there a more "proper" way of doing such a thing?

Ideally, I'd be able to just host 1 instance of Prisma with in the API that other prisma-clients can connect to so they only need the models for generating them and not also all the migrations

**Comment by nurul3101** [comment, marked as answer]
Hey @Jarco-dev 👋 

Have you considered extracting the Prisma schema and generated prisma client into an internal npm package that could be shared among all these different projects?

This should ideally solve your issue. Another possible option could be using a monorepo but that could be an overhead if you are not using it now.

**Reply by Jarco-dev** [reply]
This is not really what i ment but i think ima just stick to creating schema files in all repositories copying them over whenever they need changing but only creating migrations in 1 single location (yet to test this so not sure if it'll work)

**Reply by pantharshit00** [reply]
Hello @Jarco-dev , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

I think you should open a feature request about this. You can request a feature here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

**Reply by sanjaybora04** [reply]
i recently heard about git submodules, would that solve this issue?

**Comment by aacoimbra** [comment]
I was just facing this problem.

My solution: I’ve created a private package for my organization on github. All other projects that use the same database have this package specified in package.json.

**Reply by Jarco-dev** [reply]
Not really a comparable solution but I ended up getting into monorepo's although arguably it's not a smart idea for most situations.

**Comment by haouarihk** [comment]
i made a submodule named prisma and just imported it to the github repos

**Comment by jonahallibone** [comment]
It sounds like you probably want one API that all your apps consume. This obviates the need for copy files etc

**Reply by james-william-r** [reply]
well... when you put it like that. aha. 🤦 - this solves all my problems, thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
